[Feature Request] support latest development version of LAMMPS

### Summary

It seems that LAMMPS changed their API again. The compilation error is posted below.

### Detailed Description

```
/home/jz748/codes/deepmd-kit/source/lmp/compute_deeptensor_atom.cpp: In member function ‘virtual void LAMMPS_NS::ComputeDeeptensorAtom::init()’:
/home/jz748/codes/deepmd-kit/source/lmp/compute_deeptensor_atom.cpp:70:33: error: ‘int LAMMPS_NS::NeighRequest::half’ is protected within this context
   70 |   neighbor->requests[irequest]->half = 0;
      |                                 ^~~~
In file included from /home/jz748/codes/deepmd-kit/source/lmp/compute_deeptensor_atom.cpp:8:
/home/jz748/codes/deepmd-kit/source/build/_deps/lammps_download-src/src/neigh_request.h:54:7: note: declared protected here
   54 |   int half;    // half neigh list (set by default)
      |       ^~~~
/home/jz748/codes/deepmd-kit/source/lmp/compute_deeptensor_atom.cpp:71:33: error: ‘int LAMMPS_NS::NeighRequest::pair’ is protected within this context
   71 |   neighbor->requests[irequest]->pair = 0;
      |                                 ^~~~
In file included from /home/jz748/codes/deepmd-kit/source/lmp/compute_deeptensor_atom.cpp:8:
/home/jz748/codes/deepmd-kit/source/build/_deps/lammps_download-src/src/neigh_request.h:45:7: note: declared protected here
   45 |   int pair;    // pair is set by default
      |       ^~~~
/home/jz748/codes/deepmd-kit/source/lmp/compute_deeptensor_atom.cpp:72:33: error: ‘int LAMMPS_NS::NeighRequest::compute’ is protected within this context
   72 |   neighbor->requests[irequest]->compute = 1;
      |                                 ^~~~~~~
In file included from /home/jz748/codes/deepmd-kit/source/lmp/compute_deeptensor_atom.cpp:8:
/home/jz748/codes/deepmd-kit/source/build/_deps/lammps_download-src/src/neigh_request.h:47:7: note: declared protected here
   47 |   int compute;
      |       ^~~~~~~
/home/jz748/codes/deepmd-kit/source/lmp/compute_deeptensor_atom.cpp:73:33: error: ‘int LAMMPS_NS::NeighRequest::full’ is protected within this context
   73 |   neighbor->requests[irequest]->full = 1;
      |                                 ^~~~
In file included from /home/jz748/codes/deepmd-kit/source/lmp/compute_deeptensor_atom.cpp:8:
/home/jz748/codes/deepmd-kit/source/build/_deps/lammps_download-src/src/neigh_request.h:55:7: note: declared protected here
   55 |   int full;    // full neigh list
      |       ^~~~
/home/jz748/codes/deepmd-kit/source/lmp/compute_deeptensor_atom.cpp:74:33: error: ‘int LAMMPS_NS::NeighRequest::occasional’ is protected within this context
   74 |   neighbor->requests[irequest]->occasional = 1;
      |                                 ^~~~~~~~~~
In file included from /home/jz748/codes/deepmd-kit/source/lmp/compute_deeptensor_atom.cpp:8:
/home/jz748/codes/deepmd-kit/source/build/_deps/lammps_download-src/src/neigh_request.h:59:7: note: declared protected here
   59 |   int occasional;    // how often list is built
      |       ^~~~~~~~~~
make[2]: *** [lmp/plugin/CMakeFiles/deepmd_lmp.dir/build.make:104: lmp/plugin/CMakeFiles/deepmd_lmp.dir/__/compute_deeptensor_atom.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:419: lmp/plugin/CMakeFiles/deepmd_lmp.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

### Further Information, Files, and Links

This version:
https://github.com/lammps/lammps/releases/tag/patch_24Mar2022

--- a/source/lmp/compute_deeptensor_atom.cpp
+++ b/source/lmp/compute_deeptensor_atom.cpp
@@ -66,12 +66,16 @@ void ComputeDeeptensorAtom::init()
 {
   // need an occasional full neighbor list
 
+#if LAMMPS_VERSION_NUMBER>=20220324
+  neighbor->add_request(this, NeighConst::REQ_FULL | NeighConst::REQ_OCCASIONAL);
+#else
   int irequest = neighbor->request(this,instance_me);
   neighbor->requests[irequest]->half = 0;
   neighbor->requests[irequest]->pair = 0;
   neighbor->requests[irequest]->compute = 1;
   neighbor->requests[irequest]->full = 1;
   neighbor->requests[irequest]->occasional = 1;
+#endif
 }
 
 void ComputeDeeptensorAtom::init_list(int /*id*/, NeighList *ptr)

--- a/source/lmp/fix_dplr.cpp
+++ b/source/lmp/fix_dplr.cpp
@@ -442,7 +442,11 @@ void FixDPLR::post_force(int vflag)
   // backward communication of fcorr
   dfcorr_buff.resize(dfcorr.size());
   copy(dfcorr.begin(), dfcorr.end(), dfcorr_buff.begin());
+#if LAMMPS_VERSION_NUMBER>=20220324
+  comm->reverse_comm(this,3);
+#else
   comm->reverse_comm_fix(this,3);
+#endif
   copy(dfcorr_buff.begin(), dfcorr_buff.end(), dfcorr.begin());
   // // check and print
   // cout << "-------------------- fix/dplr: post force " << endl;

--- a/source/lmp/pair_deepmd.cpp
+++ b/source/lmp/pair_deepmd.cpp
@@ -558,7 +558,11 @@ void PairDeepMD::compute(int eflag, int vflag)
 	int rank = comm->me;
 	// std force 
 	if (newton_pair) {
-	  comm->reverse_comm_pair(this);
+#if LAMMPS_VERSION_NUMBER>=20220324
+	  comm->reverse_comm(this);
+#else
+    comm->reverse_comm_pair(this);
+#endif
 	}
 	vector<double> std_f;
 #ifdef HIGH_PREC
@@ -1035,10 +1039,14 @@ void PairDeepMD::coeff(int narg, char **arg)
 
 void PairDeepMD::init_style()
 {
+#if LAMMPS_VERSION_NUMBER>=20220324
+  neighbor->add_request(this, NeighConst::REQ_FULL);
+#else
   int irequest = neighbor->request(this,instance_me);
   neighbor->requests[irequest]->half = 0;
   neighbor->requests[irequest]->full = 1;  
   // neighbor->requests[irequest]->newton = 2;  
+#endif
   if (out_each == 1){
     int ntotal = atom->natoms;
     int nprocs = comm->nprocs;

